packages/core/src/cache/plugin.ts

```diff
@@ -0,0 +1,9 @@
+import type { Plugin, DependencyManager } from '../plugins'
+
+import { CacheRepository } from './CacheRepository'
+
+export const cachePlugin: Plugin = {
```

Comment:
I probably forgot but is there a reason why we cant make a class with one static method? I think it would improve the syntax and will help with consistency between public and non-public plugins.

Reply:
I have looked around a bit and this seems to work: (could probably be written cleaner)

```
interface MyType {
    instanceMethod(): void;
}

interface MyTypeStatic {
    new():MyType;
    staticMethod(): void;
}

function staticImplements<T>() {
    return <U extends T>(constructor: U) => {constructor};
}

@staticImplements<MyTypeStatic>()
class MyTypeClass { 
    public static staticMethod() {}
    instanceMethod() {}
}
```

playground: https://www.typescriptlang.org/play?&q=256#code/JYOwLgpgTgZghgYwgAgLIE8Aq6AOKDeAUMicqAM5hwhKoRgAWA9gCYAUAlAFzIBuTwFgG5CAX0KFQkWIhQZseAMpUwwBMiKlkICAHdOXebggitlOKoR1GrA3wHCxEmAFcaqpiGTnLASQC2OAA2EP4Q4OQAPJgAfJwaxKRQ9C5QXpEAqsgQAB6QICzkyLFsCJ6UUC4IYExQPBkcyAC8MRplIBVVNVCiIuKEAAI+agHBoeFgUUZKKmpxHIQIQXDkRdMQAMLLqxrIiSQ4LgBGQWres+rDVvTM7I344loUVDQQ1rfxD05AA

stackoverflow: https://stackoverflow.com/questions/13955157/how-to-define-static-property-in-typescript-interface